ssas/rsakeys_test.go

```diff
@@ -1,25 +1,61 @@
 package ssas
 
 import (
+	"crypto/rsa"
+	"testing"
+
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/suite"
-	"testing"
 )
 
-type KeyToolsTestSuite struct {
+const badBase64KeyPath = "../shared_files/ssas/bad_base64_test_private_key.pem"
+const notRSAKeyPath = "../shared_files/ssas/not_rsa_test_private_key.pem"
+const tooSmallKeyPath = "../shared_files/ssas/too_small_test_private_key.pem"
+const goodTestKeyPath = "../shared_files/ssas/good_test_private_key.pem"
+
+type RSAKeysTestSuite struct {
 	suite.Suite
 }
 
-func (s *KeyToolsTestSuite) SetupSuite() {
+func (s *RSAKeysTestSuite) TestReadPrivateKey_EmptyKey() {
+	pemData, err := ReadPEMFile("")
+	assert.NotNil(s.T(), err)
+	assert.Contains(s.T(), err.Error(), "no such file or directory", "expected os.open error")
+	_, err = ReadPrivateKey(pemData)
+	assert.NotNil(s.T(), err)
+	assert.Contains(s.T(), err.Error(), "empty or nil")
 }
 
-func (s *KeyToolsTestSuite) TearDownSuite() {
+func (s *RSAKeysTestSuite) TestReadPrivateKey_BadKeys() {
+	assertT := assert.New(s.T())
+	var tests = []struct {
+		path   string
+		errMsg string
+	}{
+		{badBase64KeyPath, "decode"},
+		{notRSAKeyPath, "parse RSA"},
+		{tooSmallKeyPath, "insecure key length"},
+	}
+	for _, test := range tests {
```

Comment:
This is my favorite part of the PR, with its terse code and thorough testing of keys.